src/xtgeo/grid3d/_gridprop_etc.py

```diff
@@ -38,103 +37,66 @@ def gridproperty_fromgrid(self, grid, linkgeometry=False):
 
     """
-    self._ncol = grid.ncol
-    self._nrow = grid.nrow
-    self._nlay = grid.nlay
-
-    vals = self._values
-    if vals is None:
-        vals = 0
-
-    if isinstance(vals, (int, float)):
-        dtype = np.float64
-        if self._isdiscrete:
-            dtype = np.int32
-            self._roxar_dtype = np.uint8
-
-        if isinstance(vals, int):
-            vals = int(vals)
-        else:
-            vals = float(vals)
+    self._ncol = gridlike.ncol
+    self._nrow = gridlike.nrow
+    self._nlay = gridlike.nlay
 
-        vals = np.zeros(grid.dimensions, dtype=dtype) + vals
-    else:
-        vals = vals.copy()  # do copy do avoid potensial reference issues
+    gridvalues_fromspec(self, values)
 
-    if isinstance(grid, xtgeo.grid3d.Grid):
+    if isinstance(gridlike, xtgeo.grid3d.Grid):
 
-        act = grid.get_actnum(asmasked=True)
+        act = gridlike.get_actnum(asmasked=True)
 
-        self._values = np.ma.array(vals, mask=np.ma.getmaskarray(act.values))
+        self._values = np.ma.array(self._values, mask=np.ma.getmaskarray(act.values))
 
         del act
 
         if linkgeometry:
             # assosiate this grid property with grid instance. This is not default
             # since sunch links may affect garbish collection
-            self.geometry = grid
+            self.geometry = gridlike
 
-        grid.append_prop(self)
+        gridlike.append_prop(self)
 
     else:
-        self._values = np.ma.array(vals, mask=np.ma.getmaskarray(grid.values))
-
-
-def gridproperty_fromfile(self, pfile, **kwargs):
-
-    """Make an GridProperty from file.
-
-    Args:
-        pfile (str): Name of file
-        **kwargs: Various settings
-
-    """
-    logger.debug("Import from file...")
-    fformat = kwargs.get("fformat", "guess")
+        self._values = np.ma.array(
+            self._values, mask=np.ma.getmaskarray(gridlike.values)
+        )
 
-    self.from_file(
-        pfile,
-        fformat=fformat,
-        name=self._name,
-        grid=self._geometry,
-        gridlink=kwargs.get("gridlink"),
-        date=self._date,
-        fracture=self._fracture,
-    )
 
-
-def gridproperty_fromspec(self, **kwargs):
-
-    """Make an GridProperty from kwargs spec.
+def gridvalues_fromspec(self, values):
+    """Update or set values.
 
     Args:
-        pfile (str): Name of file
-        **kwargs: Various settings
-
+        values: Values will be None, and array or a scalar
     """
-
-    # self._geometry: this is a link to the Grid instance, _only if needed_. It may
-    # potentially make trouble for garbage collection
-
-    values = kwargs.get("values", None)
-
     testmask = False
     if values is None:
         values = np.ma.zeros(self.dimensions)
-        values += 99
-        testmask = True
 
-    if values.shape != self.dimensions:
-        values = values.reshape(self.dimensions, order="C")
+        if self.dimensions == (4, 3, 5):
+            # looks like default test
```

Comment:
Is this the code recognizing it is undergoing testing? (just like vw diesel engines?)

Reply:
Will clearify (it is not under default but a default if no args are given)